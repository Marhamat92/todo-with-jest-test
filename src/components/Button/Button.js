import React from 'react'
import { View, Text, StyleSheet,TouchableOpacity } from 'react-native' 
import tw from 'twrnc';  
 

function Button({title,onClick}) {
  return (
    <TouchableOpacity onPress={onClick} testID='button-touchable' style={tw`bg-green-500 px-8 py-2 rounded-md mt-2`}>
        <Text testID='button-title' style={tw`text-white font-bold text-lg`} >{title}</Text>
    </TouchableOpacity>
  )
}

export default Button