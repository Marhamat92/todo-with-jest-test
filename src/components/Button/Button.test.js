import React from 'react'
import {render,fireEvent} from '@testing-library/react-native'
import Button from './Button'

test('should match with snapshot 1', () => { 
    const comp = render(<Button/>) 
    expect(comp).toMatchSnapshot() 
    // render the component
    // compare with snapshot,expect method is used to compare the component with snapshot
    //we can use different method instead of toMatchSnapshot
})

//below test is to check if button title is displayed correctly
test('button title should be correct',()=>{
   const testTitle = 'Button'
    const comp = render(<Button title={testTitle} />)
    const buttonText = comp.getByTestId('button-title').children[0]
    expect(buttonText).toBe(testTitle)
})

//below test is to check if button is clickable
test('should trigger onPress',()=>{
    const mockFunction = jest.fn()
    const comp  = render(<Button onClick={mockFunction} />)
    const buttonTouchable = comp.getByTestId('button-touchable')
    fireEvent.press(buttonTouchable,"press") //press is the event name,fireEvent is used to trigger the event by jest
    expect(mockFunction).toBeCalled();
})

//below test is o check button style
test('should have correct style',()=>{
    const comp = render(<Button />)
    const buttonContainerStyle = comp.getByTestId('button-touchable').props.style
    expect(buttonContainerStyle).toEqual({backgroundColor: '#00ff00'}) //we should write all css if we use tailwind otherwise,
    //we can use const to include all css or we can use vanilla css of react ,nake primary thme or secondary theme to check the style
}
)



//with this test we can see that the button is rendered correctly
//lets create example test again:

test('check number',()=>{   //this test will check if the number is equal to 10,if not it will fail
    expect (5 + 5).toBe(10)
})