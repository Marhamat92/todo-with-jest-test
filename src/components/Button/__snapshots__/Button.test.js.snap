// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#22c55e",
      "borderRadius": 6,
      "marginTop": 8,
      "opacity": 1,
      "paddingBottom": 8,
      "paddingLeft": 32,
      "paddingRight": 32,
      "paddingTop": 8,
    }
  }
  testID="button-touchable"
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 18,
        "fontWeight": "bold",
        "lineHeight": 28,
      }
    }
    testID="button-title"
  />
</View>
`;
